### PR TITLE
[DC-220] Update setup script to create snapshots

### PR DESCRIPTION
**Pull request by samanehsan**
The Data Catalog needs to create test datasets and snapshots in the Data Repo with different policy combinations. Since the `setup_ui_integration.py` script is so convenient, I made a couple of changes so we could re-use it:
- The datasets to upload are defined in a separate json file now, which has:
  - A new "schema" field so that the same schema can be used to create multiple datasets in the same environment
  - A new "upload_prefix" field for the path to the metadata ingest file
  - An optional "snapshots" field to define snapshots to create from this dataset.

Example usage for the data catalog:

1. `gcloud auth login datacatalogadmin@test.firecloud.org`
2. `python setup_ui_integration.py --host 'https://jade.datarepo-dev.broadinstitute.org' --datasets './suites/catalog_admin_datasets.json'`

Notes:  
- User credentials are in vault - see https://broadworkbench.atlassian.net/browse/DC-218 for details.
- Both test users and the Data Repo service accounts for dev, alpha, staging and prod are storage object viewers on the `catalog-testdata-useastregion` bucket containing test dataset data in the dev gcloud project.
~- We also need to add the puppeteer service account to the same policy group as the user in each environment for testing the UI~

### CODE REVIEW

**Comment by tlangs** [comment]
Glad you found it useful!

File: tools/ui_integration/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+The `setup_ui_integration.py` script can be used to easily create datasets and snapshots in the Data Repo.
```
Comment (tlangs) [inline tools/ui_integration/README.md:1]: Yay documentation!

File: tools/ui_integration/catalog_admin_datasets.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+[
```
Comment (tlangs) [inline tools/ui_integration/catalog_admin_datasets.json]: Splitting these out into separate "suites" is a great idea!

**Review by tlangs — APPROVED** [review]

File: tools/ui_integration/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+###Usage:
```
Comment (pshapiro4broad) [inline tools/ui_integration/README.md]: I think these need spaces to render properly
```suggestion
### Usage:
```

File: tools/ui_integration/setup_ui_integration.py [hunk 4]
```diff
@@ -113,8 +87,9 @@ def create_dataset(clients, dataset_to_upload, profile_id):
     dataset_name = dataset_to_upload['name']
 
     dataset = None
-    with open(os.path.join("files", dataset_name, "dataset_schema.json")) as dataset_schema_json:
+    with open(os.path.join("files", dataset_to_upload['schema'], "dataset_schema.json")) as dataset_schema_json:
```
Comment (pshapiro4broad) [inline tools/ui_integration/setup_ui_integration.py]: consistent quotes?
```suggestion
    with open(os.path.join("files", dataset_to_upload["schema"], "dataset_schema.json")) as dataset_schema_json:
```

File: tools/ui_integration/setup_ui_integration.py [hunk 5]
```diff
@@ -139,14 +152,20 @@ def main():
...
+        dataset_name = created_dataset['name']
+        output_ids = {dataset_name: {'dataset_id': created_dataset['id']}}
+        if dataset_to_upload.get('snapshots'):
+            snapshot_ids = create_snapshots(clients, dataset_to_upload['name'], dataset_to_upload['snapshots'],
+                                            profile_id)
+            output_ids[dataset_name]['snapshot_ids'] = snapshot_ids
+            outputs.append(output_ids)
+
+    output_filename = f"{args.datasets.split('.')[0]}_outputs.json"
+    with open(output_filename, 'w') as f:
```
Comment (pshapiro4broad) [inline tools/ui_integration/setup_ui_integration.py:169]: What is the output file used for? Maybe add it to the documentation as well.

**Review by pshapiro4broad — APPROVED** [review]
This looks good to me. Although, I am wondering if the catalog tests will need to create snapshots dynamically instead of using static snapshots. That's a bigger conversation though, and either way it's good to have the ability to create snapshots and policies here as well.

**Comment by pshapiro4broad** [comment]
> That's a bigger conversation though, and either way it's good to have the ability to create snapshots and policies here as well.

We talked about this during a mobbing meeting yesterday (3/30) and decided that the end-to-end UI tests need to have a consistent set of snapshots in all test environments to test different operations. The service end-to-end tests, which need to modify catalog state, will most likely need to create snapshots dynamically to avoid test collisions.

File: tools/ui_integration/setup_ui_integration.py (hunk 5 above)
Comment (samanehsan) [reply tools/ui_integration/setup_ui_integration.py:169]: The script used to print out all of the dataset ids, but I think it's easier to write the results to a file now that it includes snapshot ids as well.

**Pull request merged**